src/applications/facility-locator/components/ServiceTypeAhead.jsx

```diff
@@ -96,30 +120,69 @@ class ServiceTypeAhead extends Component {
+                        pressedEnter: false,
+                      });
+                    }
+                    this.validateServiceTypeAhead(
+                      'service-type-ahead-input',
+                      inputValue,
+                      event,
+                    );
+                  },
                   placeholder: 'Like primary care, cardiology',
                 })}
                 id="service-type-ahead-input"
                 required
               />
               {isOpen && inputValue.length >= 2 ? (
                 <div className="dropdown" role="listbox">
-                  {services
-                    .filter(specialty => this.shouldShow(inputValue, specialty))
-                    .map((specialty, index) => (
-                      <div
-                        key={this.getSpecialtyName(specialty)}
-                        {...getItemProps({
-                          item: specialty,
-                          className: this.optionClasses(
-                            index === highlightedIndex,
-                          ),
-                          role: 'option',
-                          'aria-selected': index === highlightedIndex,
-                        })}
-                      >
-                        {this.getSpecialtyName(specialty)}
-                      </div>
-                    ))}
+                  {(() => {
+                    const servicesFound = services.filter(svc =>
+                      this.shouldShow(inputValue, svc),
+                    );
+                    if (servicesFound.length > 0 && !selectedItem) {
+                      return services
+                        .filter(svc => this.shouldShow(inputValue, svc))
+                        .map((svc, index) => (
+                          <div
+                            key={svc.name}
+                            {...getItemProps({
+                              item: svc,
+                              className: this.optionClasses(
+                                index === highlightedIndex,
+                              ),
+                              role: 'option',
+                              'aria-selected': index === highlightedIndex,
+                            })}
+                          >
+                            {this.getSpecialtyName(svc)}
+                          </div>
+                        ));
+                    }
+                    if (this.state.pressedEnter && services.length > 0) {
+                      return (
+                        <div
+                          key={'not-found'}
+                          {...getItemProps({
+                            item: 'not-found',
+                            className: 'dropdown-option',
+                            role: 'option',
+                          })}
+                        >
+                          We're sorry. This service type was not found.
+                        </div>
```

Comment:
Let's try refactoring this into a function, since it's very similar to the code in lines 155-168.